src/Compilers/Core/Portable/SignatureComparer.cs

```diff
@@ -144,6 +154,30 @@ private static bool IsByRef(ImmutableArray<byte> signature, ref int position)
             }
         }
 
+        private bool MatchTypeParameterConstraints(MethodSymbol method, int index, ImmutableArray<byte> signature, ref int position)
```

Comment:
>MatchTypeParameterConstraints [](start = 21, length = 29)

If we start verifying constraints for methods, I think we should also verify them for types.